sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/retryprocessingerror/RetryProcessingErrorMLS.java

```diff
@@ -27,12 +28,27 @@ public class RetryProcessingErrorMLS<K, V> extends MessageListenerStrategy<K, V>
 
   private static final Logger LOGGER = LoggerFactory.getLogger(RetryProcessingErrorMLS.class);
   private final MessageHandler<K, V> handler;
-  private final ErrorHandler<K, V> errorHandler;
+  private final ErrorHandler<K, V> intermediateErrorHandler;
+  private final ErrorHandler<K, V> finalErrorHandler;
+  private final long maxRetryCount;
   private String consumerName;
+  private Map<TopicPartition, OffsetCounter> offsetCounters =
+      Collections.synchronizedMap(new HashMap<>());
 
-  public RetryProcessingErrorMLS(MessageHandler<K, V> handler, ErrorHandler<K, V> errorHandler) {
+  public RetryProcessingErrorMLS(
+      MessageHandler<K, V> handler, ErrorHandler<K, V> intermediateErrorHandler) {
+    this(handler, intermediateErrorHandler, Long.MAX_VALUE, null);
```

Comment:
Would it make sense to use the intermediateErrorHandler as finalErrorHandler here as well? You may run in (theoretical) NullPointerExceptions when passing null.